src/jit/lower.cpp

```diff
@@ -3118,13 +3118,17 @@ GenTree* Lowering::LowerVirtualStubCall(GenTreeCall* call)
 
         // Direct stub calls, though the stubAddr itself may still need to be
         // accesed via an indirection.        
-        GenTree* tmp = AddrGen(stubAddr);
-        tmp->gtRegNum = REG_VIRTUAL_STUB_PARAM;
-
-        tmp = Ind(tmp);
-        tmp->gtRegNum = REG_JUMP_THUNK_PARAM;
-
-        result = tmp;
+        GenTree* addr = AddrGen(stubAddr);
+        GenTree* indir = Ind(addr);
+
+#ifndef _TARGET_X86_
+        // On x86 we generate this:
+        //        call dword ptr [rel32]  ;  FF 15 ---rel32----
+        // So we don't use a register.
```

Comment:
+1. I might even flip the polarity of the comment (e.g. "On x86-64 we call through a register")